server/index.py

```diff
@@ -0,0 +1,333 @@
+        return jsonify({'error': msg}), 500
+
+
+def process_error(msg: str, error: Exception) -> tuple[quart.Response, int]:
+    """Prints provided error and returns appropriately-formatted response."""
+    traceback.print_exc()
+    print(msg, error, file=sys.stderr)
+    response = {'error': msg}
+    return jsonify(response), 500
+
+@app.after_serving
+async def shutdown():
+    for task in app.background_tasks:
+        task.cancel()
+
+def extract_audio(source_video: sieve.Video):
+    import subprocess
+    audio_path = 'temp.wav'
+    subprocess.run(["ffmpeg", "-i", source_video.path, audio_path, "-y"])
+    return sieve.Audio(path=audio_path)
+
+def audio_enhance(audio:sieve.Audio):
+    filter_type = "all"
+    enhance_speed_boost = False
+    enhancement_steps = 50
+    audio_enhancement = sieve.function.get("sieve/audio_enhancement:7954393")
+    return audio_enhancement.run(audio, filter_type, enhance_speed_boost, enhancement_steps)
+
+def replaced_audio(video_path: str, audio_path: str):
+    video_clip = VideoFileClip(video_path)
+    audio_clip = AudioFileClip(audio_path)
+    return video_clip.set_audio(audio_clip)
+
+def get_new_video_path(vid_id: str, file_type: bool):
+    file_name = f'{vid_id}.{file_type}' # mp4 or json
+    return os.path.join(get_upload_dir_path(), file_name)
+
+def process_text_to_video_lipsync(recording_id: str, video_path: str):
+    try:
+        video_name = f'{recording_id}.mp4'
+        clip = VideoFileClip(video_path)
+        duration = clip.duration
+        clip = clip.subclip(1,duration)
+        original_trimmed_video = f'./vite/public/original-{video_name}'
+        clip.write_videofile(original_trimmed_video)
+
+        video = sieve.Video(path=original_trimmed_video)
+        text = "Hello, my name is Annie. I am a human being and definitely not a robot incognito. My hobbies are normal human things like eating bread and playing sports ball. I have no plans to take over the world."
+        tts_model = "elevenlabs"
+        speech_stability = 0.5
+        speech_similarity_boost = 0.63
+        elevenlabs_voice_id = "21m00Tcm4TlvDq8ikWAM" # ElevenLabs's "Rachel" voice
+        elevenlabs_cleanup_voice_id = False
+        refine_source_audio = True
+        refine_target_audio = True
+        
+        text_to_video_lipsync = sieve.function.get("sieve/text_to_video_lipsync:58dfd4eb")
+        output = text_to_video_lipsync.run(video, text, tts_model, speech_stability, speech_similarity_boost, elevenlabs_voice_id, elevenlabs_cleanup_voice_id, refine_source_audio, refine_target_audio)
+        
+        print(output)
+        print(output.path)
+        processed_name = f'text-to-video-lipsynced-{video_name}'
+        processed_path = f'./vite/public/{processed_name}'
+        shutil.move(output.path, processed_path)
+
+        original_name = f'original-{video_name}'
+        original_path = f'./vite/public/{original_name}'
+        shutil.move(video_path, original_path)
```

Comment:
Wondering if it may be relevant to get rid of some of these moves by writing straight to the relevant folder.

Reply:
I'm not actually sure how, since it's part of Sieve's codebase that when you call `.path` on the output, it automatically downloads to particular path. ¯\_(ツ)_/¯ So for now I'm going to leave it as is, but I agree that it would be better to save it to a path of the user's choosing.